Newspack/Newspack/Controllers/ImageViewController.swift

```diff
@@ -0,0 +1,181 @@
+        imageView.alpha = 0
+    }
+
+
+    override func viewDidAppear(_ animated: Bool) {
+        super.viewDidAppear(animated)
+
+        centerImage()
+        UIView.animate(withDuration: 0.3) { [weak self] in
+            self?.imageView.alpha = 1
+        }
+    }
+
+    override func viewWillTransition(to size: CGSize, with coordinator: UIViewControllerTransitionCoordinator) {
+        super.viewWillTransition(to: size, with: coordinator)
+        coordinator.animate(alongsideTransition: { [weak self] (_) in
+            guard let scrollView = self?.scrollView else {
+                return
+            }
+            self?.scrollViewDidZoom(scrollView)
+        }, completion: nil)
+    }
+
+    func configureGestures() {
+        _ = doubleTapRecognizer
+        _ = tapRecognizer
+    }
+
+    func configureImage() {
+        guard let img = image else {
+            return
+        }
+        imageView.image = img
+        imageView.sizeToFit()
+        scrollView.contentSize = img.size
+        centerImage()
+    }
+
+    func configureFlingableViewHandler() {
+        flingableViewHandler.isActive = scrollView.zoomScale == scrollView.minimumZoomScale
+    }
+
+    func centerImage() {
+        guard let image = imageView.image else {
+            return
+        }
+        let scaleWidth = scrollView.frame.width / image.size.width
+        let scaleHeight = scrollView.frame.height / image.size.height
+        let zoomScale = min(scaleWidth, scaleHeight)
+        scrollView.minimumZoomScale = zoomScale
+        scrollView.zoomScale = zoomScale
+        scrollViewDidZoom(scrollView)
+    }
+
+}
+
+// MARK: - Actions
+
+extension ImageViewController {
+
+    @objc func handleTap(recognizer: UITapGestureRecognizer) {
+        dismiss(animated: true, completion: nil)
+    }
+
+    @objc func handleDoubleTap(recognizer: UITapGestureRecognizer) {
+        if scrollView.zoomScale > scrollView.minimumZoomScale {
+            scrollView.setZoomScale(scrollView.minimumZoomScale, animated: true)
+            return
+        }
+
+        let point = recognizer.location(in: imageView)
+        let size = scrollView.frame.size
+
+        let w = size.width / scrollView.maximumZoomScale
+        let h = size.height / scrollView.maximumZoomScale
+        let x = point.x - (w / 2)
+        let y = point.y - (h / 2)
+        let rect = CGRect(x: x, y: y, width: w, height: h)
+        scrollView.zoom(to: rect, animated: true)
+    }
+
+}
+
+// MARK: - ScrollView delegate.
+
+extension ImageViewController: UIScrollViewDelegate {
+
+    func viewForZooming(in scrollView: UIScrollView) -> UIView? {
+        return imageView
+    }
+
+    func scrollViewDidZoom(_ scrollView: UIScrollView) {
+        let w = scrollView.bounds.size.width - scrollView.contentSize.width
+        let h = scrollView.bounds.size.height - scrollView.contentSize.height
+        let x = max(w / 2, 0)
+        var y = max(h / 2, 0)
```

Comment:
Nipticky: Maybe `round` on the x/y, just in case?

Plus: `y` could also be `let` perhaps?

Reply:
Arg! I made this a var while wrestling with a positioning bug due to the status bar/nav bar being included in the initial layout pass.  I solved that issue a different way and forgot to change it back.  Well spotted.